assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py

```diff
@@ -746,3 +747,107 @@ def _prepare_artifacts_dict(self) -> Dict:
         """
         artifacts_dict = {SegmentAnythingMLflowLiterals.MODEL_DIR: self._model_dir}
         return artifacts_dict
+
+
+class AutoMLMLFlowConvertor(PyFuncMLFLowConvertor):
+    """PyFunc MLfLow convertor for AutoML models."""
+
+    MODEL_DIR = os.path.join(os.path.dirname(__file__), "automl")
+    MLflowLiteral_Model = "model"
+    MLflowLiteral_Settings = "settings"
+
+    def __init__(self, **kwargs):
+        """Initialize MLflow convertor for AutoML models."""
+        super().__init__(**kwargs)
+        if self._task not in [
+            SupportedTasks.IMAGE_CLASSIFICATION.value,
+            SupportedTasks.IMAGE_CLASSIFICATION_MULTILABEL.value
+        ]:
+            raise Exception("Unsupported task")
+
+        self._device = "cuda:0" if torch.cuda.is_available() else "cpu"
+
+
+    def _get_model_weights_classification(self) -> str:
+        """Return default model weights path.
+
+        :return: Path to default model.
+        :rtype: str
+        """
+        multilabel = False
+        if self._task == SupportedTasks.IMAGE_CLASSIFICATION_MULTILABEL.value:
+            multilabel = True
+
+        from azureml.automl.dnn.vision.classification.models import ModelFactory
+        from azureml.automl.dnn.vision.classification.common.constants import ModelNames
+        from azureml.model.mgmt.processors.pyfunc.automl.vit_classes import IMAGENET2012_CLASSES
+        import copy
+
+        model_wrapper = ModelFactory().get_model_wrapper(
+            model_name=ModelNames.VITB16R224,
+            num_classes=1000,
+            multilabel=multilabel,
+            distributed=False,
+            local_rank=0,
+            device=self._device,
+            model_state=None,
+            settings={},
+        )
+
+        specs = {
+            "multilabel": model_wrapper.multilabel,
+            "model_settings": model_wrapper.model_settings,
+            'labels': list(IMAGENET2012_CLASSES.values())
+        }
+
+        checkpoint_data = {
+            "model_name": model_wrapper.model_name,
+            "number_of_classes": model_wrapper.number_of_classes,
+            "specs": specs,
+            "model_state": copy.deepcopy(model_wrapper.state_dict()),
+        }
+
+        model_file = "/tmp/vitb16r224-3c68ea1f.pth"
+        torch.save(checkpoint_data, model_file)
+
+        return model_file
+
+
+    def get_model_signature(self) -> ModelSignature:
+        """Return MLflow model signature with input and output schema for the given input task.
+
+        :return: MLflow model signature.
+        :rtype: mlflow.models.signature.ModelSignature
+        """
+
+        from azureml.automl.dnn.vision.common.model_export_utils import _get_mlflow_signature
+        return _get_mlflow_signature(self._task)
+
+
+    def save_as_mlflow(self):
+        """Prepare model for save to MLflow."""
+        sys.path.append(self.MODEL_DIR)
```

Comment:
This is probably not needed since we are not importing anything from self.MODEL_DIR path